### PR TITLE
Fix binary docvalue_fields with padding

### LINKED ISSUE #70244: Fetch fields API inconsistent with doc_values for binary fields
We generally expect the `fields` API to be consistent with `docvalue_fields` except for `text` fields. But for `binary` fields the `docvalue_fields` API doesn't return any trailing padding characters in the base64 results. But the `fields` API returns exactly what is in the `_source`, trailing `=`s and all.

Interestingly, we can't parse the results of `docvalue_fields`. When we go to parse the binary field we call `JsonParser.getBinaryValue` which fails on base64 without the padding saying "Unexpected end of base64-encoded String: base64 variant 'MIME-NO-LINEFEEDS' expects padding (one or more '=' characters) at the end". It looks like there are some values where either `JsonParser` doesn't need the padding.

### PR DESCRIPTION
**Pull request by mayya-sharipova**
Previously docvalue_fields for binary values with paddings did not
output padding. We consider it to be a bug because: 1) es would
not be able parse these values 2) output from source filtering
and fields API is different and does output padding.

This patches fixes this by outputing padding for binary
docvalue_fields where it is present.

Closes #70244

### CODE REVIEW

File: rest-api-spec/src/main/resources/rest-api-spec/test/search/350_binary_field.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        #other formats (e.g. cbor) may not support parsing of binary
+        headers:
+            Content-Type: application/json
+        index:
+            index: test
+            id: 1
+            body:
+                binary: U29tZSBiaW5hcnkgYmxvYg==
+
+    - do:
```
Comment (nik9000) [inline rest-api-spec/src/main/resources/rest-api-spec/test/search/350_binary_field.yml:27]: I'd stick the `refresh` as a parameter on `index`. Not a big thing, but I figure you may as well save a request.

**Review by nik9000 — COMMENTED** [review]
Would you like to enable `BinaryFieldMapperTests#generateRandomInputValue` in this PR too? It was disabled because of this. I think it'd be as simple as making that method return some random binary instead of `assumeFalse`.

File: rest-api-spec/src/main/resources/rest-api-spec/test/search/350_binary_field.yml (hunk 1 above)
Comment (mayya-sharipova) [reply rest-api-spec/src/main/resources/rest-api-spec/test/search/350_binary_field.yml:28]: Addressed in 73be1d918a4e58be64be6dc4764091a090305d6c

**Comment by mayya-sharipova** [comment]
@nik9000 Thank you for your review. I've addressed your comments in 73be1d918a4e58be64be6dc4764091a090305d6c

File: server/src/test/java/org/elasticsearch/index/mapper/BinaryFieldMapperTests.java [hunk 2]
```diff
@@ -121,9 +123,12 @@ public void testStoredValue() throws IOException {
 
     @Override
     protected Object generateRandomInputValue(MappedFieldType ft) {
-        assumeFalse("We can't parse the binary doc values we send", true);
-        // AwaitsFix https://github.com/elastic/elasticsearch/issues/70244
-        return null;
+        if (rarely()) {
+            return null;
+        } else {
```
Comment (nik9000) [inline server/src/test/java/org/elasticsearch/index/mapper/BinaryFieldMapperTests.java:128]: I think it'd be a little easier to read without the `else` bit. It'd not important, but my eyes are just used to the "early return" pattern.

File: server/src/test/java/org/elasticsearch/index/mapper/BinaryFieldMapperTests.java [hunk 3]
```diff
@@ -121,9 +123,12 @@ public void testStoredValue() throws IOException {
...
+            byte[] value = randomAlphaOfLengthBetween(1, 30).getBytes(StandardCharsets.UTF_8);
```
Comment (nik9000) [inline server/src/test/java/org/elasticsearch/index/mapper/BinaryFieldMapperTests.java:129]: I think it'd be nice to also try the case where these aren't utf-8 bytes. Just some byte array of pure random stuff.

**Comment by nik9000** [comment]
Thanks for iterating! I left a request to add one more case. We have a surprising amount of code that things all byte arrays are utf-8 characters. I'm sort of hoping to create a trap for code like that.

File: server/src/test/java/org/elasticsearch/index/mapper/BinaryFieldMapperTests.java (hunk 2 above)
Comment (mayya-sharipova) [reply server/src/test/java/org/elasticsearch/index/mapper/BinaryFieldMapperTests.java:128]: @nik9000 You are super fast at reviewing :)

File: server/src/test/java/org/elasticsearch/index/mapper/BinaryFieldMapperTests.java (hunk 3 above)
Comment (mayya-sharipova) [reply server/src/test/java/org/elasticsearch/index/mapper/BinaryFieldMapperTests.java:129]: Great comment! I guess it makes sense just always use some random bytes, I guess this is what Binary field is intended for – to store random binary values.

File: server/src/test/java/org/elasticsearch/index/mapper/BinaryFieldMapperTests.java (hunk 2 above)
Comment (mayya-sharipova) [reply server/src/test/java/org/elasticsearch/index/mapper/BinaryFieldMapperTests.java]: Addressed in 3aa1286d84fe1628740d286e77e7314a92dbe9e4

File: server/src/test/java/org/elasticsearch/index/mapper/BinaryFieldMapperTests.java (hunk 3 above)
Comment (mayya-sharipova) [reply server/src/test/java/org/elasticsearch/index/mapper/BinaryFieldMapperTests.java]: Addressed in 3aa1286d84fe1628740d286e77e7314a92dbe9e4

**Review by nik9000 — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**